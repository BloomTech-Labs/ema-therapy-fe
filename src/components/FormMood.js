import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Slider } from 'antd';

const FormMood = ({ onMoodSliderChange, mood, handleView }) => {
  return (
    <>
      <div className="header">
        <Link to="/dashboard">
          <button type="button" className="back">
            &larr;
          </button>
        </Link>
        <p>How do you feel?</p>
        <button className="main-button" type="submit">
          Done
        </button>
      </div>
      <div className="inputs">
        <Slider value={mood} onChange={onMoodSliderChange} min={1} max={5} />
      </div>
      <div className="footer">
        <button
          className="main-button"
          type="button"
          onClick={() => handleView('activity-journal')}
          data-testid="next"
        >
          Next
        </button>
      </div>
    </>
  );
};

FormMood.propTypes = {
  handleView: PropTypes.func.isRequired,
  mood: PropTypes.number.isRequired,
  onMoodSliderChange: PropTypes.func.isRequired,
};

export default FormMood;
