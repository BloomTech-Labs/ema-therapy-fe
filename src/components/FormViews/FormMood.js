import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { Slider, Icon } from 'antd';
import NextButton from './NextButton';
import moodToString from '../../utils/moodToString';
import useStandalone from '../../hooks/useStandalone';
import happyPlant from '../../assets/plants-png/plant-happy.png';
import normalPlant from '../../assets/plants-png/plant-normal.png';
import sadPlant from '../../assets/plants-png/plant-sad.png';
import reallyHappyPlant from '../../assets/plants-png/plant-really-happy.png';
import unhappyPlant from '../../assets/plants-png/plant-unhappy.png';

function FormMood({ onMoodSliderChange, mood, handleView, stopEditing }) {
  const history = useHistory();
  const isStandalone = useStandalone();

  const getPlant = (m) => {
    let plant;
    if (m === 1) plant = unhappyPlant;
    if (m === 2) plant = sadPlant;
    if (m === 3) plant = normalPlant;
    if (m === 4) plant = happyPlant;
    if (m === 5) plant = reallyHappyPlant;
    return plant;
  };

  return (
    <>
      <div className="header">
        <Icon
          className="back-btn"
          type="left"
          onClick={() =>
            stopEditing ? stopEditing() : history.push('/dashboard')
          }
        />
        <p>
          How do you <br />
          feel?
        </p>
      </div>
      <MoodWrapper isStandalone={isStandalone}>
        <img src={getPlant(mood)} alt="happy plant" />

        <p>{moodToString(mood)}</p>
        <Slider
          min={1}
          max={5}
          value={mood}
          tooltipVisible={false}
          onChange={onMoodSliderChange}
        />
      </MoodWrapper>
      <div className="footer">
        <NextButton
          data-testid="next"
          onClick={() => handleView('anxiety-sleep')}
        >
          Next
        </NextButton>
      </div>
    </>
  );
}

FormMood.propTypes = {
  handleView: PropTypes.func.isRequired,
  mood: PropTypes.number.isRequired,
  onMoodSliderChange: PropTypes.func.isRequired,
  stopEditing: PropTypes.func,
};

FormMood.defaultProps = {
  stopEditing: null,
};

export default FormMood;

const MoodWrapper = styled.div`
  padding: 0 20px;
  display: flex;
  flex-direction: column;

  p {
    text-align: center;
    color: #fcb924;
    font-size: 16px;
    text-transform: capitalize;
    margin-bottom: 0;
  }

  img {
    margin-bottom: 15px;
    height: 350px;
    width: 228px;
    align-self: center;
  }

  @media only screen and (max-width: 375px) {
    img {
      height: ${(props) => (props.isStandalone ? '300px' : '250px')};
      width: ${(props) => (props.isStandalone ? '195px' : '163px')};
    }
  }
`;
