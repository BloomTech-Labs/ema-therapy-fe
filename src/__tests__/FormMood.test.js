import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import FormMood from '../components/FormViews/FormMood';
import 'jest-styled-components';

describe('FormMood component', () => {
  test('renders correctly', () => {
    expect(
      render(
        <Router>
          <FormMood
            onMoodSliderChange={() => {}}
            handleView={() => {}}
            handleSubmit={() => {}}
            isSubmitting
            mood={5}
          />
        </Router>,
      ),
    ).toMatchSnapshot();
  });
  test('next button calls onClick handleView', () => {
    const handleViewMock = jest.fn();
    const { getByTestId } = render(
      <Router>
        <FormMood
          onMoodSliderChange={() => {}}
          handleView={handleViewMock}
          handleSubmit={() => {}}
          isSubmitting
          mood={5}
        />
      </Router>,
    );
    fireEvent.click(getByTestId('next'));
    expect(handleViewMock).toHaveBeenCalledWith('anxiety-sleep');
  });
});
