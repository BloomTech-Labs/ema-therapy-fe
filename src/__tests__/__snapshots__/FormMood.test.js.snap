// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormMood component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  padding: 0 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 p {
  text-align: center;
  color: #fcb924;
  font-size: 16px;
  text-transform: capitalize;
  margin-bottom: 0;
}

.c0 img {
  margin-bottom: 15px;
  height: 350px;
  width: 228px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media only screen and (max-width:375px) {
  .c0 img {
    height: 250px;
    width: 163px;
  }
}

<body>
    <div>
      <div
        class="header"
      >
        <i
          aria-label="icon: left"
          class="anticon anticon-left"
          style="font-size: 22px; color: rgb(156, 217, 221);"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </i>
        <p>
          How do you 
          <br />
          feel?
        </p>
        <button
          class="ant-btn ant-btn-loading"
          style="background: rgb(255, 255, 255); color: rgb(137, 210, 214); font-size: 12px; border-radius: 10px; width: 70px; height: 30px; margin-right: -7px; margin-top: 6px;"
          type="button"
        >
          <i
            aria-label="icon: loading"
            class="anticon anticon-loading"
          >
            <svg
              aria-hidden="true"
              class="anticon-spin"
              data-icon="loading"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
              />
            </svg>
          </i>
          <span>
            Done
          </span>
        </button>
      </div>
      <div
        class="c0"
      >
        <img
          alt="happy plant"
          src="plant-really-happy.png"
        />
        <p>
          happy
        </p>
        <div
          class="ant-slider"
        >
          <div
            class="ant-slider-rail"
          />
          <div
            class="ant-slider-track"
            style="left: 0%; width: 100%;"
          />
          <div
            class="ant-slider-step"
          />
          <div
            aria-disabled="false"
            aria-valuemax="5"
            aria-valuemin="1"
            aria-valuenow="5"
            class="ant-slider-handle"
            role="slider"
            style="left: 100%; transform: translateX(-50%);"
            tabindex="0"
          />
          <div
            class="ant-slider-mark"
          />
        </div>
      </div>
      <div
        class="footer"
      >
        <button
          class="ant-btn"
          data-testid="next"
          style="background: rgb(0, 145, 122); color: rgb(255, 255, 255); font-size: 18px; border-radius: 10px; width: 167px; height: 55px; box-shadow: 0px 4px 4px #CCE8E4;"
          type="button"
        >
          <span>
            Next
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <i
        aria-label="icon: left"
        class="anticon anticon-left"
        style="font-size: 22px; color: rgb(156, 217, 221);"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="left"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
          />
        </svg>
      </i>
      <p>
        How do you 
        <br />
        feel?
      </p>
      <button
        class="ant-btn ant-btn-loading"
        style="background: rgb(255, 255, 255); color: rgb(137, 210, 214); font-size: 12px; border-radius: 10px; width: 70px; height: 30px; margin-right: -7px; margin-top: 6px;"
        type="button"
      >
        <i
          aria-label="icon: loading"
          class="anticon anticon-loading"
        >
          <svg
            aria-hidden="true"
            class="anticon-spin"
            data-icon="loading"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
            />
          </svg>
        </i>
        <span>
          Done
        </span>
      </button>
    </div>
    <div
      class="sc-bdVaJa eLbXML"
    >
      <img
        alt="happy plant"
        src="plant-really-happy.png"
      />
      <p>
        happy
      </p>
      <div
        class="ant-slider"
      >
        <div
          class="ant-slider-rail"
        />
        <div
          class="ant-slider-track"
          style="left: 0%; width: 100%;"
        />
        <div
          class="ant-slider-step"
        />
        <div
          aria-disabled="false"
          aria-valuemax="5"
          aria-valuemin="1"
          aria-valuenow="5"
          class="ant-slider-handle"
          role="slider"
          style="left: 100%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="ant-slider-mark"
        />
      </div>
    </div>
    <div
      class="footer"
    >
      <button
        class="ant-btn"
        data-testid="next"
        style="background: rgb(0, 145, 122); color: rgb(255, 255, 255); font-size: 18px; border-radius: 10px; width: 167px; height: 55px; box-shadow: 0px 4px 4px #CCE8E4;"
        type="button"
      >
        <span>
          Next
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
